core/function.cc

```diff
@@ -684,10 +684,10 @@ toRowInfo() const
 
 FunctionOutput
 FunctionApplier::
-apply(const FunctionContext & input) const
-{
+apply(const SqlRowScope & outer, const FunctionContext & input) const
```

Comment:
I would add some documentation to outer parameter, describing exactly why it's needed and what should be passed in.